### PR TITLE
Refactor `payjoin-cli` v1, v2 modules

### PR DESCRIPTION
**Pull request by DanGould**
Note that I used the async-trait crate because we target msrv 1.63.0

msrv 1.70 allows async traits out of the box, so when we target that we can remove the dependency

### CODE REVIEW

**Comment by DanGould** [comment]
@jbesraa Separating `main.rs` from library code should let us write cli integration tests to test things like retry v1/v2 and ohttp-relay details a lot more easily without dealing with command-line parsing

File: payjoin-cli/src/main.rs [hunk 1]
```diff
@@ -18,22 +24,14 @@ async fn main() -> Result<()> {
...
-            #[cfg(not(feature = "v2"))]
-            app.send_payjoin(bip21, fee_rate_sat_per_vb).await?;
         }
         Some(("receive", sub_matches)) => {
             let amount =
                 sub_matches.get_one::<String>("AMOUNT").context("Missing AMOUNT argument")?;
-            #[cfg(feature = "v2")]
-            let is_retry = matches.get_one::<bool>("retry").context("Could not read --retry")?;
-            #[cfg(feature = "v2")]
+            let is_retry = matches.get_one::<bool>("retry").unwrap_or(&false);
```
Comment (jbesraa) [inline payjoin-cli/src/main.rs:33]: are you planning to take this out in a different PR?

**Review by jbesraa — APPROVED** [review]
Concept ACK.

Really like the new design with the trait and file separation! makes the code much more comprehensive and readable!

File: payjoin-cli/src/main.rs (hunk 1 above)
Comment (DanGould) [reply payjoin-cli/src/main.rs:33]: Yes, I'll follow up with it's removal. This was focused on the module separation only

**Pull request merged**